### PR TITLE
[feat](cloud) S3 storage vault support property `use_path_style`

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
example 
```
  CREATE STORAGE VAULT IF NOT EXISTS demo_vault
    PROPERTIES (
      "type"="S3",
      ...
      "use_path_style" = "true"
  );
```
doc https://github.com/apache/doris-website/pull/1344

### CODE REVIEW

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java [hunk 1]
```diff
@@ -334,6 +334,11 @@ public static Cloud.ObjectStoreInfoPB.Builder getObjStoreInfoPB(Map<String, Stri
             // S3 Provider properties should be case insensitive.
             builder.setProvider(Provider.valueOf(properties.get(S3Properties.PROVIDER).toUpperCase()));
         }
+
+        if (properties.containsKey(PropertyConverter.USE_PATH_STYLE)
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:338]: can it be modified by `alter vault` stmt?

Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:338]: > can it be modified by `alter vault` stmt?

now, don't supported, maybe will support in the future

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java [hunk 2]
```diff
@@ -334,6 +335,18 @@ public static Cloud.ObjectStoreInfoPB.Builder getObjStoreInfoPB(Map<String, Stri
             // S3 Provider properties should be case insensitive.
             builder.setProvider(Provider.valueOf(properties.get(S3Properties.PROVIDER).toUpperCase()));
         }
+
+        if (properties.containsKey(PropertyConverter.USE_PATH_STYLE)) {
+            String value = properties.get(PropertyConverter.USE_PATH_STYLE);
+            Preconditions.checkArgument(!Strings.isNullOrEmpty(value), "use_path_style cannot be empty");
+            Preconditions.checkArgument(value.equalsIgnoreCase("true")
+                    || value.equalsIgnoreCase("false"), "Invalid use_path_style value: %s", value);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:343]: "Invalid use_path_style value: %s, only 'true' or 'false' is acceptable"

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java [hunk 3]
```diff
@@ -334,6 +335,18 @@ public static Cloud.ObjectStoreInfoPB.Builder getObjStoreInfoPB(Map<String, Stri
...
+            if (properties.get(PropertyConverter.USE_PATH_STYLE).equalsIgnoreCase("true")) {
+                builder.setUsePathStyle(true);
+            } else {
+                builder.setUsePathStyle(false);
+            }
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:348]:
```suggestion
            builder.setUsePathStyle(value.equalsIgnoreCase("true"));
```

Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:348]: maybe alter set `use_path_style` from true to false

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java (hunk 2 above)
Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java]: > "Invalid use_path_style value: %s, only 'true' or 'false' is acceptable"

resolved

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by Yukang-Lian — APPROVED** [review]

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run p0

**Comment by SWJTU-ZhangLei** [comment]
run cloud_p0

**Comment by SWJTU-ZhangLei** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**